mapswipe_workers/tests/integration/test_firebase_db_rules.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+
+# import re
+
+# from mapswipe_workers.auth import firebaseDB
+# from mapswipe_workers.definitions import CustomError
+
+
+class TestFirebaseDBRules(unittest.TestCase):
+    def setUp(self):
+        # set up 4 users with specified roles
+        # normal app user
+
+        # team member
+
+        # project manager
+
+        # project manager and team member
+
+        pass
+
+    def tearDown(self):
+        # tear down up 4 users with specified roles
+        pass
+
+    def test_normal_user(self):
```

Comment:
@laurentS just a quick on the structure of the tests: not sure which way round is nicer? test based on user role or based on endpoint?

just seeing this now it seems that by endpoint might be easier to understand.

Reply:
Based on my comment [here](https://github.com/mapswipe/python-mapswipe-workers/issues/392#issuecomment-658192603) I would write a function that pings each endpoint in turn, with a user passed as argument, and call it for each of the users you're interested in. Actually, from my perspective, you don't even care about the results, you just want to make sure you're not getting errors, so you could probably have a function that tests for error, given an endpoint, and then use that in the first function I described above?

```python
def test_endpoint(user, endpoint):
   # run the query and check for error

def test_all_app_endpoints(user):
    test_endpoint(user, endpoint1)
    test_endpoint(user, endpoint2)
    # ...
```